ergo/distributions/logistic_mixture.py

```diff
@@ -3,151 +3,66 @@
 """
 from dataclasses import dataclass
 import itertools
-from typing import Optional, Sequence
+from typing import Sequence
 
 from jax import nn
 import jax.numpy as np
 import numpy as onp
 
-from ergo.conditions import Condition
-from ergo.scale import Scale
 import ergo.static as static
-from ergo.utils import minimize
 
 from .logistic import Logistic
 from .mixture import Mixture
+from .optimizable import Optimizable
 
 
 @dataclass
-class LogisticMixture(Mixture):
+class LogisticMixture(Mixture, Optimizable):
     components: Sequence[Logistic]
     probs: Sequence[float]
 
     def logpdf(self, x):
-        return static.logistic_mixture_logpdf(self.to_params(), x)
-
-    def grad_logpdf(self, x):
-        return static.logistic_mixture_grad_logpdf(self.to_params(), x)
-
-    def to_params(self):
         nested_params = [
             [c.loc, c.scale, weight] for c, weight in zip(self.components, self.probs)
         ]
-        return np.array(list(itertools.chain.from_iterable(nested_params)))
+        opt_params = np.array(list(itertools.chain.from_iterable(nested_params)))
+        return static.logistic_mixture_logpdf(opt_params, x)
 
     @classmethod
-    def from_params(cls, params):
-        structured_params = params.reshape((-1, 3))
+    def from_params(cls, fixed_params, opt_params, traceable=True):
```

Comment:
I'm confused by the fact that this method takes `fixed_params` and `traceable` but doesn't use them

maybe a Python idiom I'm not familiar with?

Reply:
The reason is to keep the interface the same as in `Optimizable`. `HistogramDist` also implements `Optimizable` and does use those arguments.